lib/eventFactory.js

```diff
@@ -453,50 +455,88 @@ async function convertStages(config) {
             }
         });
 
+        newStage.eventId = eventId;
         newStage.jobIds = jobIds;
+
         delete newStage.jobs; // extra field from yaml parser
 
+        // Check for setup and teardown
+        const setupJobName = `${STAGE_PREFIX}${key}:setup`;
+        const teardownJobName = `${STAGE_PREFIX}${key}:teardown`;
+        const setupJobExists = Object.keys(jobs).find(j => j === setupJobName || j === `~${setupJobName}`);
+        const teardownJobExists = Object.keys(jobs).find(j => j === teardownJobName || j === `~${teardownJobName}`);
+
+        if (setupJobExists) {
+            newStage.setup = [pipelineJobs.find(j => j.name === setupJobExists).id];
+        }
```

Comment:
How do we handle the use when `setup` does not exist and needs to be created?

- `setup is explicitly specified in the `screwdriver.yaml` config 
- `setup is not specified in the `screwdriver.yaml` config

Reply:
Same for `teardown`